framework/src/partitioner/PetscExternalPartitioner.C

```diff
@@ -110,21 +139,30 @@ PetscExternalPartitioner::_do_partition(MeshBase & mesh, const unsigned int n_pa
   nj = 0;
   for (auto & row : _dual_graph)
   {
-    if (_apply_side_weight)
-      neighbor = 0;
+    auto & elem = _local_id_to_elem[local_elem_id];
+    unsigned int n_neighbors = 0;
 
-    for (auto adj : row)
+    for (auto neighbor : elem->neighbor_ptr_range())
     {
-      j[nj] = adj;
-      if (_apply_side_weight)
+      // Skip boundary sides since they do not connect to
+      // anything.
+      if (neighbor != nullptr && neighbor->active())
       {
-        auto & elem = _local_id_to_elem[local_elem_id];
-        values[nj] = computeSideWeight(*elem, neighbor);
-        neighbor++;
-      }
+        j[nj] = row[n_neighbors++];
 
-      nj++;
+        if (_apply_side_weight)
+        {
+          auto side = elem->which_neighbor_am_i(neighbor);
```

Comment:
`which_neighbor_am_i()` is actually expensive (it's a search).  Let's go back to looping over sides.